fixed measures (#8)
leanprover-community/liquid: src/measures.tex

  \lean{real_measures}
  \leanok
  Let $0 < p' < 1$ be a real number, and let $S$ be a finite set.
  Denote by $\Rm(S)$ the real vector space
\[

> \Rm(S)=\left\{\sum_{s\in S}a_{s} \text{ such that } a_{s}\in \R\right\}

Reviewer: I think it's good to include a formal generator in the notation:
```suggestion
\Rm(S)=\left\{\sum_{s\in S}a_{s}[s] \text{ such that } a_{s}\in \R\right\}
```
Also elsewhere, see the diff.

\]

> endowed with the $\ell^{p'}$-norm $\Vert \sum_{s\in S}a_s\Vert_{\ell^{p'}}=\sum_{s\in S}\lvert a_s\rvert ^{p'}$. We refer to its elements as (real) $\ell^{p'}$-measures on $S$.

Reviewer: ```suggestion
endowed with the $\ell^{p'}$-norm $\Vert \sum_{s\in S}a_s[s]\Vert_{\ell^{p'}}=\sum_{s\in S}\lvert a_s\rvert ^{p'}$. We refer to its elements as (real) $\ell^{p'}$-measures on $S$.
```

\end{definition}

> For every finite set $S$, the space $\Rm(S)$ can be written as the filtered colimit

Reviewer: I think I would just call it a union
```suggestion
For every finite set $S$, the space $\Rm(S)$ can be written as the union
```

Author: Well, on this I would disagree. In my opinion, in the language of LTE it is more natural to consider it as a colimit, expecially because we need to use its relationship with limits. But of course you decide at the end of the day, so if you stick to your view I will be happy to change.

Reviewer: Ok, fair enough. How about
```suggestion
For every finite set $S$, the space $\Rm(S)$ can be written as the filtered colimit (in this case, a union)
```

\[

> \Rm(S)=\varinjlim_{c>0}\Rm(S)_{\leq c}

Reviewer: ```suggestion
\Rm(S)=\bigcup_{c>0}\Rm(S)_{\leq c}
```

Author: This is of course related to the above, but if you opt for my version we should revert all $\bigcup$ to $\varinjlim$ everywhere in the document.

Reviewer: Agreed. I'm fine with using `\varinjlim`.

Author: OK, should I modify it everywhere (e.g. at the beginning of the proof of Lemma 2.4.11)?

	\Lm_{r'}(S)_{A,\leq c}=\bigl\{\sum_{s\in S,n\in A}a_{n,s}T^n[s] \mid a_{n,s}\in \Z, \sum_{n\in A, s\in S}\lvert a_{n,s}\rvert r^n\leq c\bigr\}.
\]
This defines a profinite topology on $\Lm_{r'}(S)_{\leq c}$, for each $c>0$.
\end{definition}
Passing to the limit, we obtain the Laurent measures on an arbitrary profinite set $S$ as follows. 
%\begin{definition}
% \label{laurent-measures}
% \lean{laurent_measures}
% \leanok
%  Let $0 < r' < 1$ be a real number, and let $S$ be a profinite set with 

> Starting with a presentation $S=\varprojlim S_i$, where all the $S_i$ are finite, define for all $c>0$ the space $\Lm_{r'}(S)_{\leq c}$ denote the limit

Reviewer: This sentence is a bit mangled. The "define" and "denote" aren't mixing well with each other.

%  The result follows from Lemma~\ref{theta.limit_y}.
%\end{proof}

%Now fix $0< \xi<1$ and let $x\in\R_{\geq 0}$. For simplicity, denote by $\Z(\!(T)\!)_{r'}$ the group $\Lm_{r'}(\ast)$.
%One of the key results proven in \cite[\S 6]{Analytic} is the surjectivity and continuity of the following map.

\begin{definition}
  \label{theta}
  \lean{theta.ϑ}
  \uses{laurent-measures}
  \leanok
  Let $0<\xi <1$ be a real number. Denote by $\vartheta_{\xi}\colon \Z(\!(T)\!)_{r'}\rightarrow \R$ the evaluation map
  \[
    \sum a_nT^n\longmapsto \sum a_n\xi^n.
  \]
induced by $T\mapsto \xi$.
\end{definition}

One of the key results proven in \cite[\S 6]{Analytic} is the surjectivity and continuity of the map $\vartheta_\xi$, together with a proof that its kernel is principal for every $0<\xi<1$. To state this result, we begin with the following lemma.

% \begin{definition}
%  \label{theta.y}
%  \lean{theta.y}
%  \leanok
%	, namely the greatest integer that is less or equal than $\ast$.
%\end{definition}

\begin{lemma}

>   \label{theta.surjective}

Reviewer: This breaks the blueprint software. LaTeX labels cannot contain `.` for some reason.

\begin{proof}
  \uses{theta-y}
  \leanok
  Given $n\in\N$ and $x\in\R$, set
	\[
		Y_\xi(x, n)=
		\begin{cases}
			x&\text{ if } n = 0 \\
			Y_\xi(x,n-1) - \left\lfloor \frac{Y_\xi(x,n-1)}{\xi^{n-1}} \right\rfloor \xi^{n-1} & \text{ if } n \ge 1
		\end{cases}
	\]
where $\lfloor \ast \rfloor$ denotes the floor of $\ast$. Consider the power series
  \[
    F(T)=\sum_{n\geq 0}\left\lfloor \frac{Y_\xi(x,n)}{\xi^n}\right\rfloor T^n\in\Z[\![T]\!].
  \]
	The fact that $0<\xi<1$ ensures that the series converges on the open unit disk, so it actually belongs to $\Z(\!(T)\!)_{r'}$ and it can be proven that $F(\xi)=x$.
%  By Proposition~\ref{theta.has_sum_x}, the series converges on the open unit disks,
%  so it belongs to $\Z((T))_r$ and it specializes to $x$ when evaluated at $T=\xi$.
\end{proof}
The next lemma allows to compare the seemingly decorrelated topologies on the spaces of \emph{real} and of \emph{Laurent} measures:
\begin{lemma}

> 	\label{theta.continuous}

Reviewer: ```suggestion
	\label{theta-continuous}
```
See the other comment.

	\uses{theta}
	\uses{laurent-measures-fin}
	\uses{real-measures}
	\lean{ses.continuous_θ_c}
	\leanok
%Let $0<p'<1$ be a real number, let $S$ be a finite set, and let $r'$ denote $\left(\frac{1}{2}\right)^{p'}$. 
Fix a finite set $S$, real numbers $0<\xi,p'<1$ and set $r'=(\xi)^{p'}$. The map $\vartheta_\xi$ of Definition~\ref{theta} is continuous when endowing $\Lm_{r'}(S)$ with the topology defined in Definition~\ref{laurent-measures-fin} and $\Rm(S)$ with the topology defined in Definition~\ref{real-measures}
%\[
%\theta_{S}\colon \Lm_{r'}(S)\longrightarrow \Rm(S)
%\]
%induced by evaluating $T$ at $1/2$ is continuous
\end{lemma}
\begin{proof}
  \leanok
  Omitted for now, but done in Lean.
\end{proof}

Given any finite set $S$, the map $\vartheta_{\xi}$ defines a map $\vartheta_{\xi,S}\colon \Lm_{r'}(S)\longrightarrow \Rm(S)$ by extending $\vartheta_{\xi}$ componentwise. For the purpose of this project, the special case where $\xi=1/2$ is enough, and we write $\theta_S$ from now on to denote $\vartheta_{\xi,S}$. 

%By taking projective limits, we obtain a map $\vartheta_{S}$ for every profinite $S$.

\begin{theorem}

>   \label{theta.exact_finite}

Reviewer: Idem. Maybe also elsewhere?

Author: I have checked all of them in the measures.tex file.
